statsmodels/tsa/stattools.py

```diff
@@ -1876,15 +1888,23 @@ def arma_order_select_ic(
     model_kw = {} if model_kw is None else model_kw
     fit_kw = {} if fit_kw is None else fit_kw
     y_arr = array_like(y, "y", contiguous=True)
-    for ar in ar_range:
-        for ma in ma_range:
-            mod = _safe_arma_fit(y_arr, (ar, 0, ma), model_kw, trend, fit_kw)
-            if mod is None:
-                results[:, ar, ma] = np.nan
-                continue
-
-            for i, criteria in enumerate(ic):
-                results[i, ar, ma] = getattr(mod, criteria)
+
+    def get_ic_for_order(ar: int, ma: int, criteria: str):
+        mod = _safe_arma_fit(y_arr, (ar, 0, ma), model_kw, trend, fit_kw)
+        if mod is None:
+            return np.nan
+        return getattr(mod, criteria)
+    
+    def fit_order(ar, ma):
+        return [get_ic_for_order(ar, ma, criteria) for criteria in ic]
```

Comment:
this is not a good loop.
`get_ic_for_order`  fits the model.
So you are fitting the model for each ic criterion separately, repeating the same work each time.
This should get all criteria from one `fit`.